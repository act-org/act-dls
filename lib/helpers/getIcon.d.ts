/**
 * @prettier
 */
import * as React from 'react';
declare const getIcon: (selector: string) => React.FC<any>;
export default getIcon;
//# sourceMappingURL=getIcon.d.ts.map