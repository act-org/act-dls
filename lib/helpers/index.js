"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "getIcon", {
  enumerable: true,
  get: function get() {
    return _getIcon["default"];
  }
});
Object.defineProperty(exports, "search", {
  enumerable: true,
  get: function get() {
    return _search["default"];
  }
});
Object.defineProperty(exports, "sort", {
  enumerable: true,
  get: function get() {
    return _sort["default"];
  }
});

var _getIcon = _interopRequireDefault(require("./getIcon"));

var _search = _interopRequireDefault(require("./search"));

var _sort = _interopRequireDefault(require("./sort"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }