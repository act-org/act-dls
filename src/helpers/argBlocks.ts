


interface PropType {
    defaultValue: any,
    description: string,
    name: string,
    required: boolean,
    type: {
        name: string
    }
}
interface DocGenType {
    __docgenInfo: {
        description: string;
        displayName: string;
        props: Record<string, PropType>
    }
}

function cleanPropType(prop: PropType) {
    if (prop && prop.type && prop.type.name) {
        prop.type.name = prop.type.name.replace(' | undefined', '');
    }
}

const playgroundGroup = 'Playground';
const eventsGroup = 'Events';
const ariaGroup = "Aria";
const elseGroup = 'Everything Else';

/**
 * Create a dictionary of StoryBook arg types.  Be default the type and control will
 * be automatically inferred and this helper does an ok job of setting up a default UI.
 * 
 * This method also groups the parameters specified in the args param in the Playground group
 * and then puts all methods that start with on* in the events group, all props starting with 
 * aria* in the aria group and everthing else in an Everthing Else group
 * @param args The arguments to put in the Playground group
 * @param type The base type of the arguments.  If passed will group all arguments by some basic rules
 */
export function PlayGround(args: Record<string, any>, type?: any) {
    for (const key in args) {
        if (!key.startsWith('on')) {
            args[key]["table"] = { category: playgroundGroup }
        }
    }
    if (type) {
        const docType = type as unknown as DocGenType;
        const props = docType.__docgenInfo?.props;
        for (const key in props) {
            cleanPropType(props[key]);
            let category: string = elseGroup;
            if (key.startsWith('aria')) {
                category = ariaGroup;
            } else if (key.startsWith('on')) {
                category = eventsGroup;
            }
            args[key] = Object.assign(
                {
                    table: {
                        category
                    }
                },
                args[key]
            );
        }
    }
    return args;
}