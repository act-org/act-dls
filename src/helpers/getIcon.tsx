/**
 * @prettier
 */

import * as React from 'react';
import { constant } from 'lodash';
import {
  CloudDownload,
  KeyboardArrowDown,
  KeyboardArrowLeft,
  KeyboardArrowRight,
  KeyboardArrowUp,
  Sync,
} from '@material-ui/icons';

const getIcon = (selector: string): React.FC<any> => {
  switch (selector) {
    case 'general.arrow.down':
      return KeyboardArrowDown;
    case 'general.arrow.left':
      return KeyboardArrowLeft;
    case 'general.arrow.right':
      return KeyboardArrowRight;
    case 'general.arrow.up':
      return KeyboardArrowUp;
    case 'general.download':
      return CloudDownload;
    case 'general.refresh':
      return Sync;
    default:
      return constant(<div>?</div>);
  }
};

export default getIcon;
