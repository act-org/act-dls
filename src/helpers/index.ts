/**
 * @prettier
 */

export { default as mergeClasses } from './mergeClasses';
export { default as search } from './search';
export { default as sort } from './sort';
