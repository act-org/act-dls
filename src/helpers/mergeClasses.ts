/**
 * @prettier
 */

import clsx from 'clsx';
import { mergeWith } from 'lodash';

type Classes = any;

const mergeClasses = (classes1: Classes, classes2: Classes): Classes =>
  mergeWith(classes1 || {}, classes2 || {}, (className1, className2): string =>
    clsx(className1, className2),
  );

export default mergeClasses;
