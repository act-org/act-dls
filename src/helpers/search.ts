/**
 * @prettier
 */

import { get, toString } from 'lodash';

function search<T>(docs: T[], properties: string[], value: string): T[] {
  return docs.filter(doc =>
    properties.some((property): boolean => {
      const currentValue = get(doc, property);
      if (doc !== null && currentValue !== null) {
        return toString(currentValue)
          .toLowerCase()
          .includes(value.trim().toLowerCase());
      }
      return false;
    }),
  );
}

export default search;
