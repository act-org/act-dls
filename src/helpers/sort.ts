/**
 * @prettier
 */

import { isString } from 'lodash';

import SORT_DIRECTION_TYPES from '~/constants/sortDirectionTypes';
import { SortObject } from '~/types';

type SortFunction = (itemA: any, itemB: any) => number;

function sort({ dataKey, direction }: SortObject): SortFunction {
  return ({ [dataKey]: itemA }, { [dataKey]: itemB }): number => {
    if (itemA === null || itemB === null) return 0;

    let normalizedItemA: any = itemA;
    let normalizedItemB: any = itemB;

    if (isString(itemA)) {
      normalizedItemA = itemA.trim().toLowerCase();
    }

    if (isString(itemB)) {
      normalizedItemB = itemB.trim().toLowerCase();
    }

    if (normalizedItemA < normalizedItemB) {
      return direction === SORT_DIRECTION_TYPES.ASC ? -1 : 1;
    }

    if (normalizedItemA > normalizedItemB) {
      return direction === SORT_DIRECTION_TYPES.ASC ? 1 : -1;
    }

    return 0;
  };
}

export default sort;
