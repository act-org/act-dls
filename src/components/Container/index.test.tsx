/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import * as React from 'react';

import { render, THEMES } from '~/helpers/test';

import { Container } from '.';

describe('Container', () => {
  test.each(THEMES)('%s theme matches the snapshot', theme => {
    const { container } = render(<Container>children</Container>, theme);

    expect(container).toMatchSnapshot();
  });
});
