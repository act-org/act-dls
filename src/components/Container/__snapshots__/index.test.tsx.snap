// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container ACT theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    Hey
  </div>
</div>
`;

exports[`Container ACT_ET theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    Hey
  </div>
</div>
`;
